topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsConfiguration.java

```diff
@@ -58,7 +67,34 @@ public void start(Config config, Properties kafkaProperties) {
             String storeTopic = config.get(Config.STORE_TOPIC);
             String storeName = config.get(Config.STORE_NAME);
 
-            long timeoutMillis = config.get(Config.STALE_RESULT_TIMEOUT);
+            // check if entry topic has the right configuration
+            Admin admin = Admin.create(kafkaProperties);
+            DescribeClusterResult clusterResult = admin.describeCluster();
+            int clusterSize = clusterResult.nodes().get().size();
+            Set<String> topics = admin.listTopics().names().get();
+            if (topics.contains(storeTopic)) {
+                TopicDescription topicDescription = admin.describeTopics(Collections.singleton(storeTopic)).values().get(storeTopic).get();
+                int rf = topicDescription.partitions().get(0).replicas().size();
```

Comment:
should we check that it's really just one partition and not more than that?

Reply:
If it's more than one partition, which replicas size do we take then? Max, avg, min, ... ?

Reply:
Kafka doesn't really have a first-class concept of RF even though it's routinely talked about. Different partitions of the same topic can be replicated to different numbers of brokers. And indeed, that might happen (transiently) even in a well-managed cluster, e.g. increasing RF, but doing it in a batched way. For our purposes here using the min, or just using partition 0 would make sense, but I guess using the min might be preferable since it detects the case of inhomogeneous topic.